### DISCUSSION TITLE
docsearch cannot search anything

### DISCUSSION THREAD

**Discussion by Sun-ZhenXing** [vuepress/core#1270] [Q&A] [answered]
As the title, find a similar question here <https://juejin.cn/post/7160948190375018504>.

test `default-theme` & `hope-theme`, remove `facetFilters` works, any better solutions?

**Comment by Mister-Hope** [comment]
We had no solutions for people who REFUSE to read docs.

**Comment by Sun-ZhenXing** [comment]
@Mister-Hope 

I have read <https://v2.vuepress.vuejs.org/zh/reference/plugin/docsearch.html>，<https://theme-hope.vuejs.press/guide/feature/search.html#use-vuepress-plugin-docsearch>, not less than 10 times.

Not for the following reasons:
- `attributesForFaceting: ["type", "lang"]`

If this is an obvious problem, how can it not be obvious in the document and take several hours to find it?

**Reply by Mister-Hope** [reply]
Have you tried deleting index? The `initialIndexSettings` works while the index is not generated. That warning can be clearly found on Algolia, and the default crawler config contain that.

**Reply by Sun-ZhenXing** [reply]
I'm not sure what you're talking about, but I've deleted index.

I also tested the `lang` attr, but not work.

```js
      recordExtractor: ({ helpers }) => {
        return helpers.docsearch({
          recordProps: {
            lvl0: {
              selectors: ".sidebar-heading.active",
              defaultValue: "Documentation",
            },
            lvl1: ".theme-hope-content h1",
            lvl2: ".theme-hope-content h2",
            lvl3: ".theme-hope-content h3",
            lvl4: ".theme-hope-content h4",
            lvl5: ".theme-hope-content h5",
            lvl6: ".theme-hope-content h6",
            content: ".theme-hope-content p, .theme-hope-content li",
            tags: {
              defaultValue: ["blog"],
            },
            lang: {
              defaultValue: ["zh-CN"],
            },
          },
          indexHeadings: true,
          aggregateContent: true,
        });
      },
```

Before that, it was available, which may be the problem you said.
But what should I do now? I kind of want to consider giving up.

I have read various documents provided by algolia:
- <https://docsearch.algolia.com/docs/crawler#why-do-i-see-two-algolia-apps-in-my-dashboard>
- <https://docsearch.algolia.com/docs/record-extractor/#indexing-content-for-faceting>

If you have any documents related to this, please also provide them. Thank you very much

**Reply by Mister-Hope** [reply]
Docs about what I am talking: https://www.algolia.com/doc/tools/crawler/apis/configuration/initial-index-settings/#how-to-use


This is waline crawler config:

```js
new Crawler({
  appId: "REPLACED",
  apiKey: "REPLACED",
  rateLimit: 8,
  startUrls: ["https://waline.js.org/"],
  sitemaps: ["https://waline.js.org/sitemap.xml"],
  ignoreCanonicalTo: false,
  exclusionPatterns: [],
  discoveryPatterns: ["https://waline.js.org/**"],
  schedule: "at 20:00 every 1 day",
  actions: [
    {
      indexName: "waline",
      pathsToMatch: ["https://waline.js.org**/**"],
      recordExtractor: ({ $, helpers }) => {
        return helpers.docsearch({
          recordProps: {
            lvl0: {
              selectors: ".sidebar-heading.active",
              defaultValue: "Documentation",
            },
            lvl1: ".theme-hope-content h1",
            lvl2: ".theme-hope-content h2",
            lvl3: ".theme-hope-content h3",
            lvl4: ".theme-hope-content h4",
            lvl5: ".theme-hope-content h5",
            lvl6: ".theme-hope-content h6",
            content: ".theme-hope-content p, .theme-hope-content li",
          },
          indexHeadings: true,
        });
      },
    },
  ],
  initialIndexSettings: {
    waline: {
      attributesForFaceting: ["type", "lang"],
      attributesToRetrieve: ["hierarchy", "content", "anchor", "url"],
      attributesToHighlight: ["hierarchy", "hierarchy_camel", "content"],
      attributesToSnippet: ["content:10"],
      camelCaseAttributes: ["hierarchy", "hierarchy_radio", "content"],
      searchableAttributes: [
        "unordered(hierarchy_radio_camel.lvl0)",
        "unordered(hierarchy_radio.lvl0)",
        "unordered(hierarchy_radio_camel.lvl1)",
        "unordered(hierarchy_radio.lvl1)",
        "unordered(hierarchy_radio_camel.lvl2)",
        "unordered(hierarchy_radio.lvl2)",
        "unordered(hierarchy_radio_camel.lvl3)",
        "unordered(hierarchy_radio.lvl3)",
        "unordered(hierarchy_radio_camel.lvl4)",
        "unordered(hierarchy_radio.lvl4)",
        "unordered(hierarchy_radio_camel.lvl5)",
        "unordered(hierarchy_radio.lvl5)",
        "unordered(hierarchy_radio_camel.lvl6)",
        "unordered(hierarchy_radio.lvl6)",
        "unordered(hierarchy_camel.lvl0)",
        "unordered(hierarchy.lvl0)",
        "unordered(hierarchy_camel.lvl1)",
        "unordered(hierarchy.lvl1)",
        "unordered(hierarchy_camel.lvl2)",
        "unordered(hierarchy.lvl2)",
        "unordered(hierarchy_camel.lvl3)",
        "unordered(hierarchy.lvl3)",
        "unordered(hierarchy_camel.lvl4)",
        "unordered(hierarchy.lvl4)",
        "unordered(hierarchy_camel.lvl5)",
        "unordered(hierarchy.lvl5)",
        "unordered(hierarchy_camel.lvl6)",
        "unordered(hierarchy.lvl6)",
        "content",
      ],
      distinct: true,
      attributeForDistinct: "url",
      customRanking: [
        "desc(weight.pageRank)",
        "desc(weight.level)",
        "asc(weight.position)",
      ],
      ranking: [
        "words",
        "filters",
        "typo",
        "attribute",
        "proximity",
        "exact",
        "custom",
      ],
      highlightPreTag: '<span class="algolia-docsearch-suggestion--highlight">',
      highlightPostTag: "</span>",
      minWordSizefor1Typo: 3,
      minWordSizefor2Typos: 7,
      allowTyposOnNumericTokens: false,
      minProximity: 1,
      ignorePlurals: true,
      advancedSyntax: true,
      attributeCriteriaComputedByMinProximity: true,
      removeWordsIfNoResults: "allOptional",
    },
  },
});
```

And here is waline vuepress config:

https://github.com/walinejs/waline/blob/668ac8dd0aa4b8328920f309d0ef9850384bfc9a/docs/src/.vuepress/config.ts#L30-L77

Everything is working fine, if you still inisit you are meeting problems, contact algolia support.

**Reply by Sun-ZhenXing** [reply]
Thank you for your help, I have informed algolia by email and the problem has been solved.

**Reply by Mister-Hope** [reply]
So what's the actual problem?

**Reply by Sun-ZhenXing** [reply]
I have made a number of mistakes, regardless of the impact on the outcome, and I list them.

1. When this warning appears, the crawler needs to remove all the index with or without changes, the system may change it automatically, and the index should be generated by the system instead of specifying one
2. make sure that all your filters appear, if they do not appear is a problem with the settings, you need to turn on filtering in the index settings for tags, etc. Make sure that *Searchable attributes* and *Attributes for faceting* are set correctly
3. when there is any problem, you should delete the index and re-run the crawler to generate a new index

<img width="496" alt="image" src="https://user-images.githubusercontent.com/44517244/221732774-d017d5b4-20a4-4899-a9c6-c3379625e127.png">

**Comment by Sun-ZhenXing** [comment, marked as answer]
> I have made a number of mistakes, regardless of the impact on the outcome, and I list them.
> 
> 1. When this warning appears, the crawler needs to remove all the index with or without changes, the system may change it automatically, and the index should be generated by the system instead of specifying one
> 2. make sure that all your filters appear, if they do not appear is a problem with the settings, you need to turn on filtering in the index settings for tags, etc. Make sure that _Searchable attributes_ and _Attributes for faceting_ are set correctly
> 3. when there is any problem, you should delete the index and re-run the crawler to generate a new index
> 
> <img alt="image" width="496" src="https://user-images.githubusercontent.com/44517244/221732774-d017d5b4-20a4-4899-a9c6-c3379625e127.png">

So the problem is solved, thank you for your attention.

**Reply by Mister-Hope** [reply]
> Have you tried deleting index? The initialIndexSettings works while the index is not generated. That warning can be clearly found on Algolia, and the default crawler config contain that.

So basically this is the actual problem and solution. Again here for anyone who ran into this.

The default index generated by algolia after apply should be correct, and probably he is providing a wrong `initialIndexSettings ` when creating his new `xxx.yyy`. So I am actually not expecting anyone who can easily run into this.
